ctypesgen/__main__.py

```diff
@@ -28,114 +27,84 @@ def find_names_in_modules(modules):
     return names
 
 
-def option_callback_W(option, opt, value, parser):
```

Comment:
IMO, how this bit of code works is a bit convoluted.
I am not a Python expert. The following is mainly based on my knowledge of C and Makefiles.

Here is my understanding.
Please review and confirm if this is correct...

1. `ctypesgen` will be invoked with `CFLAGS`
    - In an existing Makefile, it is easy to invoke it similar to where `gcc` is being invoked already. 
2. CFLAGS include various definitions/undefs of macros and also linker flags (`-Wl,...`).
3. The `ctypesgen` logic here  appears to be:
    a. Using `option_callback_W()` look for anything following a `-Wl,`
    b. If it is NOT one of `-L`, or `-R`, or `--rpath` then complain.
    c. Else process them by now faking it as if the the options were passed in without the preceding `-Wl,`.
    d. Subsequently, process any of the 3 flags passed in using `option_callback_libdir()`
    
  i.e. invoking `ctypesgen` with a `-Wl,-L<path>`
  would result in `ctypesgen` including `<path>` in `compile_libdirs` and `runtime_libdirs`. 
_(and similarly `-Wl,-R<path>` or `-Wl,--rpath<path>`)_

Am not sure the update continues to handle this,
as there is no `parser.add_argument()` that handles a `-Wl,`.

Right?

Reply:
I haven't dived into what arguments exactly gcc receives, but all I can tell is that this code as-is could not work, as explained [in the other thread](https://github.com/ctypesgen/ctypesgen/pull/161/files#r1032772959).
Unless you can prove by example that the previous state *did* work, I would prefer if you could discuss this separately in an issue or make a PR if you know how to fix it.

Reply:
> I would prefer if you could discuss this separately in an issue or make a PR if you know how to fix it.

OK. Sounds like a good way to divide this porting problem. 👍 

I suppose we have captured enough details of the analysis of this code in this thread and [the other thread](https://github.com/ctypesgen/ctypesgen/pull/161/files#r1032772959) for future reference if/when we need to handle this "linker flags" parsing in future.

Do you think it might be helpful 
to add a permalink to these 2 comment threads 
as an inline comment at the new line-83?
_(just a thought. if not, that's fine too)_